File: CompilationTest/StructureTest1/results/StructureTest1.cpp
#include <OneLang-Core-v0.1/one.hpp>
#include <iostream>
#include <string>

template<typename T>
class MyList {
  public:
    one::vec<T> items;

  private:
};

class Item {
  public:
    int offset = 5;
    std::string str_test = std::string("test") + std::string("test2");
    std::string str_constr = std::string("constr");

    Item(std::string str_constr) {
        this->str_constr = str_constr;
    }

  private:
};

class Container {
  public:
    one::sp<MyList<Item>> item_list;
    one::sp<MyList<std::string>> string_list;

    void method0() {
    }
    
    std::string method1(std::string str) {
        return str;
    }

  private:
};

int main()
{
    std::cout << std::string("ok") << std::endl;
    return 0;
}

File: CompilationTest/StructureTest1/results/StructureTest1.swift
class MyList<T> {
  var items: [T]?
}

class Item {
  var offset: Int = 5
  var strTest: String = "test" + "test2"
  var strConstr: String = "constr"

  init(strConstr: String) {
      self.strConstr = strConstr
  }
}

class Container {
  var itemList: MyList<Item>?
  var stringList: MyList<String>?

  func method0() -> Void {
  }

  func method1(str: String) -> String {
      return str
  }
}

print("ok")